csvcubed/csvcubed/readers/configdeserialiser.py

```diff
@@ -184,197 +128,14 @@ def _metadata_from_dict(config: dict) -> "CatalogMetadata":
-#             return QbColumn(
-#                 column_name,
-#                 new_dimension,
-#                 csv_column_value_url_template,
-#             )
-#         # elif maybe_attribute_uri is not None and maybe_property_value_url is not None:
-#         #     if (
-#         #         maybe_attribute_uri
-#         #         == "http://purl.org/linked-data/sdmx/2009/attribute#unitMeasure"
-#         #     ):
-#         #         distinct_unit_uris = [
-#         #             uritemplate.expand(
-#         #                 maybe_property_value_url, {csv_safe_column_name: u}
-#         #             )
-#         #             for u in set(pandas_input_to_columnar_str(column_data))
-#         #         ]
-#         #         dsd_component = QbMultiUnits(
-#         #             [ExistingQbUnit(u) for u in distinct_unit_uris]
-#         #         )
-#         #     else:
-#         #         dsd_component = ExistingQbAttribute(maybe_attribute_uri)
-#         #
-#         #     return QbColumn(column_name, dsd_component, maybe_property_value_url)
-#         elif maybe_unit_uri is not None and maybe_measure_uri is not None:
-#             measure_component = ExistingQbMeasure(maybe_measure_uri)
-#             unit_component = ExistingQbUnit(maybe_unit_uri)
-#             observation_value = QbSingleMeasureObservationValue(
-#                 measure=measure_component,
-#                 unit=unit_component,
-#                 data_type=maybe_data_type or "decimal",
-#             )
-#             return QbColumn(column_name, observation_value)
-#         elif maybe_data_type is not None:
-#             return QbColumn(
-#                 column_name, QbMultiMeasureObservationValue(maybe_data_type)
-#             )
-#         else:
-#             raise Exception(f"Unmatched column definition: {col_config}")
-#
-#     elif isinstance(col_config, bool) and col_config:
-#         return SuppressedCsvColumn(column_name)
-#     else:
-#         # If not a known/expected type/value (or is a string), treat it as a dimension.
-#         maybe_description: Optional[str] = None
-#         if isinstance(col_config, str):
-#             maybe_description = col_config
-#
-#         new_dimension = NewQbDimension.from_data(
-#             column_name, column_data, description=maybe_description
-#         )
-#         return QbColumn(column_name, new_dimension)
-
-
-def get_cube_from_data(csv_path: Path, data: PandasDataTypes) -> QbCube:
+
+def get_cube_from_data(csv_path: Path) -> QbCube:
     """
     Given a Pandas DataFrame, it resolves each column to a column based on naming conventions
     The cube must have dimension, observation, units and measures columns
     The csv_path arg is used for the Cube title
     """
+    data: pd.DataFrame = pd.read_csv(csv_path)
     if data.shape[0] < 2:
         # Must have 2 or more rows, a heading row and a data row
         raise ValueError("The csv data did not contain sufficient rows to continue")
```

Comment:
Let's change this message to something like `CSV input must contain header row and at least one row of data`.

Reply:
Message updated, code refactored to '_read_and_check_csv(csv_path)' to for the csv read and checks, following changes to enable  mixed config/convention columns.